### PR TITLE
Use preact-compat in place of React

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

replaces react with preact-compat (till we can ditch compat)

## What is the value of this and can you measure success?

fewer bytes 💃
- ~`40kb` gzipped

## Does this affect other platforms - Amp, Apps, etc?

profile

## Tested in CODE?

- [x] tested dotcom
- [x] tested profile

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,8 @@
     "lodash-node": "^2.4.1",
     "object-fit-videos": "^1.0.3",
     "ophan-tracker-js": "1.3.9",
+    "preact": "^8.2.9",
+    "preact-compat": "^3.18.0",
```
Comment (SiAdcock) [inline package.json]: Don't forget to uninstall `react` and `react-dom` 🎉

Comment (sndrs) [reply package.json]: yeah v good point, cheers!

Comment (sndrs) [reply package.json]: all gone 👏

**Review by SiAdcock — APPROVED** [review]
Can't quite believe it was so straightforward! Nice one @sndrs

**Comment by sndrs** [comment]
all your work getting it jsx/react14/5'd i reckon

**Review by walaura — APPROVED** [review]
if privacy prefs works, LGTM!

is there anything special we have to do to get rid of `compat`?

**Comment by sndrs** [comment]
i believe some `refs` stuff, but not looked too deep here. some webpack config stuff too

**Comment by SiAdcock** [comment]
We should go through and systematically delete these [lint rule overrides](https://github.com/guardian/frontend/blob/master/static/src/javascripts/.eslintrc.js#L28), that will probably get us most of the way there

**Pull request merged**